scanner.cpp

```diff
@@ -0,0 +1,30 @@
+#include <string.h>
+#include <map>
+
+class Scanner {
+  private:
+    std::map<string, string> Grammar;
+
+  public:
+    Scanner() {
```

Comment:
We might want to initialize the Grammar map in the initialization list (there should be `{{...}}` syntax to do this). That might be a little cleaner and more performant.

Reply:
Also I might call it "Terminals" instead of "Grammar" but that's a nit.

Reply:
@domfarolino Yeah, my C++ is probably terrible.  Also new to C++ 14/17.

Reply:
This article (http://thispointer.com/different-ways-to-initialize-an-unordered_map/) might be a good reference for the initialization. I don't think it is C++ 14/17 specific (at least hopefully!)